frontend/openchat-client/src/openchat.ts

```diff
@@ -3743,12 +3744,24 @@ export class OpenChat extends OpenChatAgentWorker {
     }
 
     getCurrentUser(): Promise<CurrentUserResponse> {
-        return this.sendRequest({ kind: "getCurrentUser" }).then((response) => {
-            if (response.kind === "created_user") {
-                userCreatedStore.set(true);
-                selectedAuthProviderStore.init(AuthProvider.II);
-            }
-            return response;
+        return new Promise((resolve, reject) => {
+            let resolved = false;
+            this.sendStreamRequest({ kind: "getCurrentUser" })
+                .subscribe((user) => {
+                    if (user.kind === "created_user") {
+                        userCreatedStore.set(true);
+                        selectedAuthProviderStore.init(AuthProvider.II);
```

Comment:
Why is this hardcoded to II?

Reply:
It's a default that gets set if there is nothing in local storage